MetaMask/metamask-extension
# Change SwapsController onNetworkStateChange to networkDidChange

## **Description**

Fixes bug with the Web3Provider being reinstantiated with the provider proxy before it was updated to point to the new network. This is achieved by changing the listened event from `stateChange` to `networkDidChange`

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21904

## **Manual testing steps**

1. Change your network to Linea
2. Restart extension
3. Open extension
4. Change network to Mainnet
5. Open swaps
6. Swap 1 WETH for ETH 
7. Should no longer error

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
